### PR TITLE
fix(core,platform): radio a11y improvements

### LINKED ISSUE #11281: [Sourcing-2][Accessibility: ACC-264.1][WIP] Radio Button: Label and group label not read.
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
Component having issue:https://sap.github.io/fundamental-ngx/#/core/radio
radio button:
The radio buttons should always be associated to its individual and group label.

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
All latest version

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Using Screen reader can check by implementing similar case. Just use any Radio Button. Radio buttons will not be associated to its label both label  and also its group label header by screen reader
#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations. 
Yes we have got this bug from Accessibility expert only.

#### Did you check the documentation and the API?
Yes

#### Is there anything else we should know?

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes #11281 

Various improvements to radio button accessibility, including some changes to examples. I think some of the screenshots from the linked issue are dated, so this PR addresses things a little differently

### CODE REVIEW

File: libs/platform/table/components/table-view-settings-dialog/grouping/grouping.component.html [hunk 1]
```diff
@@ -17,9 +17,15 @@ <h4 fd-title [headerSize]="4">{{ 'platformTable.groupDialogHeader' | fdTranslate
     <fd-dialog-body disablePaddings>
         <ul fd-list [selection]="true">
             <li fd-list-group-header>
-                <span fd-list-title>{{ 'platformTable.groupDialogGroupOrderHeader' | fdTranslate }}</span>
+                <span fd-list-title id="group-dialog-group-order-header">{{
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-view-settings-dialog/grouping/grouping.component.html:20]: in my opinion, this id should be auto generated and then assigned as aria-describedby to the list items (also automatically).

File: libs/docs/core/list-byline/examples/list-byline-selection-example/list-byline-selection-example.component.html [hunk 2]
```diff
@@ -11,26 +11,44 @@ <h4>With Checkboxes</h4>
...
         </div>
     </li>
     <li fd-list-item [selected]="selectionValue === 'radio3'">
-        <fd-radio-button name="radio1" id="radio3" value="radio3" [(ngModel)]="selectionValue"></fd-radio-button>
+        <fd-radio-button
+            name="radio1"
+            id="radio3"
+            value="radio3"
+            [(ngModel)]="selectionValue"
+            aria-describedby="with-radio-buttons"
```
Comment (InnaAtanasova) [inline libs/docs/core/list-byline/examples/list-byline-selection-example/list-byline-selection-example.component.html:50]: I think it's more appropriate in this case to have `aria-labelledby` then `aria-describedby`
Screenshot 2024-10-03 at 12 30 19 PM

File: libs/platform/table/components/table-view-settings-dialog/grouping/grouping.component.html [hunk 3]
```diff
@@ -43,9 +49,15 @@ <h4 fd-title [headerSize]="4">{{ 'platformTable.groupDialogHeader' | fdTranslate
 
         <ul fd-list [selection]="true">
             <li fd-list-group-header>
-                <span fd-list-title>{{ 'platformTable.groupDialogGroupByHeader' | fdTranslate }}</span>
+                <span fd-list-title id="group-dialog-group-by-header">{{
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-view-settings-dialog/grouping/grouping.component.html:52]: I think this should also be `[attr.id]="groupOrderHeaderId"`

File: libs/platform/table/components/table-view-settings-dialog/sorting/sorting.component.html [hunk 4]
```diff
@@ -16,9 +16,15 @@ <h4 fd-title [headerSize]="4">{{ 'platformTable.sortDialogHeader' | fdTranslate
     <fd-dialog-body disablePaddings>
         <ul fd-list [selection]="true">
             <li fd-list-group-header>
-                <span fd-list-title>{{ 'platformTable.sortDialogSortOrderHeader' | fdTranslate }}</span>
+                <span fd-list-title id="sort-dialog-sort-order-header">{{
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-view-settings-dialog/sorting/sorting.component.html:19]: This is still hard-coded

File: libs/platform/table/components/table-view-settings-dialog/sorting/sorting.component.html [hunk 5]
```diff
@@ -16,9 +16,15 @@ <h4 fd-title [headerSize]="4">{{ 'platformTable.sortDialogHeader' | fdTranslate
...
+                    'platformTable.sortDialogSortOrderHeader' | fdTranslate
+                }}</span>
             </li>
-            <li fd-list-item [selected]="direction === SORT_DIRECTION.ASC">
+            <li
+                fd-list-item
+                [selected]="direction === SORT_DIRECTION.ASC"
+                aria-describedby="sort-dialog-sort-order-header"
```
Comment (InnaAtanasova) [inline libs/platform/table/components/table-view-settings-dialog/sorting/sorting.component.html:26]: same as above

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: libs/platform/table/components/table-view-settings-dialog/grouping/grouping.component.html (hunk 3 above)
Comment (mikerodonnell89) [reply libs/platform/table/components/table-view-settings-dialog/grouping/grouping.component.html:52]: good catch

**Review by InnaAtanasova — APPROVED** [review]

**Pull request closed without merging**